services/api/tests/test_oauth_state_csrf.py

```
@@ -93,20 +93,29 @@ def test_verify_accepts_state_within_ttl(_oauth_mod, monkeypatch):
+    """#91 (greptile P1, re-verified during #563 disposition): the original
+    assertion OR'd the signature comparison with a rand-component comparison,
+    which is near-always true regardless of the signature (rand is
+    independently random per call) - a broken/constant-signature HMAC would
+    still have passed. Hold rand+ts fixed across both calls so the
+    signature comparison is the only thing that can make this pass."""
     monkeypatch.setenv("GITHUB_APP_WEBHOOK_SECRET_SSM", "/grug/test-webhook-secret")
     import importlib
     import auth.github_oauth as mod
     importlib.reload(mod)
 
+    monkeypatch.setattr(mod.secrets, "token_urlsafe", lambda n: "fixed-rand")
```

Ruff (0.15.21)

[warning] Unused lambda argument: `n`

(ARG005)